File: Week1Day5/fibonacci/fibonacci.py
# DSAA Week 1 Day 5, 7/18/2024

# fibonacci sequence 

# this program is bugged its not workign idk why
def fibonacci(n):
    if n == 0 or 1:
        if n == 0:
            return 0

        else:
            return 1
    
    else:
        return fibonacci(n-1) + fibonacci(n-2)

s = fibonacci(5)
print (s)

File: Week1/Week1Day5/fibonacci/fibonacci.java
// DSAA Week 1 Day 5, 7/18/2024

// Objective: write a fibonacci sequence

package fibonacci;

public class fibonacci 
{
    public static void main (String[] args)
    {
        // max is 43
        double result = fibi(43);
        System.out.println(result);
    }

    // fibonacci 
    public static double fibi (int num)
    {
        if (num == 0 || num == 1)
        {
            return 1;
        }
        
        else if (num > 0)
        {
            return fibi(num - 1) + fibi(num - 2);
        }

        // teminate
        else
        {
            return 0;
        }
    }
}
